### PR TITLE
Updated to SDK 5.1.1-beta

### PR DESCRIPTION
**Pull request by lotaked**
Refactored configuration and element initialization, updated dependencies, and improved code maintainability.

- Removed hardcoded values in `App.Debug.config` and `App.Release.config`.
- Added assembly binding redirects in `App.config`.
- Enhanced `StartConnector` method in `App.xaml.cs` with additional settings and removed `AssemblyResolve` event handler.
- Simplified `RenderStyle` initialization and updated element properties in `CreateExchangeHelper.cs`.
- Replaced `Application.ClearBusyState` and `Application.SetBusyState` with `ClearBusyMessage` and `ShowBusyMessage`.
- Updated exchange data handling logic and made `UpdateLocalExchange` and `SelectElements` methods asynchronous.
- Upgraded various `Autodesk.DataExchange` libraries and other dependencies in `SampleConnector.csproj` and `packages.config`.
- Added new references and removed outdated ones, including `System.Net.Http`.
- Updated `SampleConnectorUnitTests.csproj` with new dependencies and removed old error conditions.
- Added new assembly binding redirects in `app.config`.

These changes improve configuration management, enhance element initialization, and ensure the project uses the latest dependencies while enforcing coding standards.

### CODE REVIEW

File: src/App.config [hunk 1]
```diff
@@ -101,6 +101,14 @@
 				<assemblyIdentity name="System.Diagnostics.DiagnosticSource" publicKeyToken="cc7b13ffcd2ddd51" culture="neutral" />
 				<bindingRedirect oldVersion="0.0.0.0-7.0.0.2" newVersion="7.0.0.2" />
 			</dependentAssembly>
+			<dependentAssembly>
+				<assemblyIdentity name="System.Reflection.TypeExtensions" publicKeyToken="b03f5f7f11d50a3a" culture="neutral" />
+				<bindingRedirect oldVersion="0.0.0.0-4.1.1.0" newVersion="4.1.1.0" />
+			</dependentAssembly>
+			<dependentAssembly>
+				<assemblyIdentity name="PubnubPCL" publicKeyToken="dc66f52ce6619f44" culture="neutral" />
+				<bindingRedirect oldVersion="0.0.0.0-6.19.5.0" newVersion="6.19.5.0" />
```
Comment (rinku-thakur) [inline src/App.config:110]: @lotaked , we are using PubnunPCL 6.17.0, wondering why it is redirecting to 6.19.5.

File: src/SampleConnector.csproj [hunk 2]
```diff
@@ -224,6 +238,9 @@
       <HintPath>packages\protobuf-net.Core.3.0.101\lib\net461\protobuf-net.Core.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="PubnubPCL, Version=6.19.5.0, Culture=neutral, PublicKeyToken=dc66f52ce6619f44, processorArchitecture=MSIL">
```
Comment (rinku-thakur) [inline src/SampleConnector.csproj:241]: Hi @lotaked , same here. PubnubPCL 6.17.0 is used in FDX SDK, Are our packages requesting 6.19.5?

**Review by rinku-thakur — APPROVED** [review]
Other than PubNubPCL, it looks good to me.

File: src/App.config (hunk 1 above)
Comment (lotaked) [reply src/App.config]: Thank you for pointing that out! It was automatically updated by Visual Studio when updating the SDK. I've corrected it now.

File: src/SampleConnector.csproj (hunk 2 above)
Comment (lotaked) [reply src/SampleConnector.csproj]: Thank you for pointing that out! It was automatically updated by Visual Studio when updating the SDK. I've corrected it now.

**Pull request merged**